Fresh `keryx new` app fails its own `bun run lint` (tsc) on v0.38.0: leader: string vs string|null + scaffold tsconfig/deps

## Summary

A freshly scaffolded app (`bunx keryx@latest new`) does not pass its own `bun run lint`
(`tsc && biome check .`) out of the box on **keryx 0.38.0**. There is one genuine type error in
keryx's published source, plus a few `keryx new` scaffold issues that make `tsc` fail before you
write any application code.

## Environment

- keryx: `0.38.0`
- bun: `1.3.13`
- typescript: `6.0.3` (keryx declares `typescript: ^6.0.3`)
- OS: macOS (darwin 25.5)

## 1. Type error in published source (main bug)

`initializers/actionts.ts` → `taskDetails()`:

```ts
const details: {
  queues: { [key: string]: any };
  workers: { [key: string]: any };
  stats: { [key: string]: any };
  leader: string;                       // line 666
} = { queues: {}, workers: {}, stats: {}, leader: "" };
// ...
details.leader = await api.resque.queue.leader();   // line 679
```

`api.resque.queue.leader()` returns `string | null` (node-resque), but `details.leader` is typed
`string`, so any consumer running `tsc` under `strict` gets:

```
node_modules/keryx/initializers/actionts.ts(679,5): error TS2322: Type 'string | null' is not assignable to type 'string'.
  Type 'null' is not assignable to type 'string'.
```

Because keryx ships its `.ts` source and `keryx/index.ts` side-effect-imports every initializer,
consumers can't avoid type-checking this file (`skipLibCheck` only covers `.d.ts`).

**Suggested fix:** type `leader` as `string | null`, or coalesce: `details.leader = (await api.resque.queue.leader()) ?? "";`

## 2. `keryx new` scaffold issues (v0.38.0)

A fresh scaffold's `bun run lint` fails before #1 is even reached:

- **No `typescript` devDependency** in the generated `package.json`, yet `lint` runs `tsc` →
  `tsc: command not found`.
- **`tsconfig.json` sets `"types": ["bun-types"]`** while devDeps include `@types/bun` →
  `error TS2688: Cannot find type definition file for 'bun-types'`. Should be `"types": ["bun"]`
  (or add a `bun-types` dep).
- **`drizzle-kit: ^0.20.18`** is stale relative to `drizzle-orm ^0.45`. keryx's
  `initializers/db.ts` → `generateMigrations()` builds `dbCredentials: { url }`, which does not
  type-check against drizzle-kit 0.20's `Config`:
  `error TS2353: Object literal may only specify known properties, and 'url' does not exist in type '{ wranglerConfigPath: string; dbName: string; }'`.
  Bumping `drizzle-kit` to a current version (matching `drizzle-orm`) resolves it.

## Reproduction

```bash
bunx keryx@latest new backend --yes
cd backend && cp .env.example .env
bun add -d typescript          # otherwise: tsc: command not found
bun run lint                   # fails (bun-types, drizzle-kit, then leader)
```

## Suggested fixes

- **Source:** `leader: string | null` (or `?? ""`) in `taskDetails()`.
- **Scaffold:** add a `typescript` devDependency; set `"types": ["bun"]`; bump `drizzle-kit` to
  match `drizzle-orm`.

--- a/packages/keryx/__tests__/scaffold.test.ts
+++ b/packages/keryx/__tests__/scaffold.test.ts
@@ -85,6 +85,24 @@ describe("scaffoldProject", () => {
     expect(pkg.dependencies["drizzle-orm"]).toBeDefined();
     expect(pkg.dependencies["drizzle-zod"]).toBeDefined();
     expect(pkg.devDependencies["drizzle-kit"]).toBeDefined();
+    // `lint`/`format` run `tsc`, so a fresh app must ship its own typescript (#510)
+    expect(pkg.devDependencies.typescript).toBeDefined();
+  });
+
+  test("tsconfig types match the @types/bun package (#510)", async () => {
+    await scaffoldProject("ts-project", targetDir("ts-project"), {
+      includeDb: true,
+      includeExample: true,
+    });
+
+    const tsconfig = JSON.parse(
+      fs.readFileSync(
+        path.join(targetDir("ts-project"), "tsconfig.json"),
+        "utf-8",
+      ),
+    );
+    // `@types/bun` provides the `bun` types package, not `bun-types`
+    expect(tsconfig.compilerOptions.types).toEqual(["bun"]);
   });
 
   test("skips db files when includeDb is false", async () => {

--- a/packages/keryx/initializers/actionts.ts
+++ b/packages/keryx/initializers/actionts.ts
@@ -676,7 +676,7 @@ export class Actions extends Initializer {
       details.queues[queue] = { length: length };
     }
 
-    details.leader = await api.resque.queue.leader();
+    details.leader = (await api.resque.queue.leader()) ?? "";
 
     return details;
   };

--- a/packages/keryx/package.json
+++ b/packages/keryx/package.json
@@ -1,6 +1,6 @@
 {
   "name": "keryx",
-  "version": "0.38.1",
+  "version": "0.38.2",
   "module": "index.ts",
   "type": "module",
   "license": "MIT",

--- a/packages/keryx/util/scaffold.ts
+++ b/packages/keryx/util/scaffold.ts
@@ -173,7 +173,7 @@ export function generateTsconfigContents(): string {
           target: "ESNext",
           module: "ESNext",
           moduleResolution: "bundler",
-          types: ["bun-types"],
+          types: ["bun"],
           strict: true,
           skipLibCheck: true,
           noEmit: true,
@@ -333,6 +333,7 @@ export async function scaffoldProject(
         devDependencies: {
           "@biomejs/biome": "^2.4.8",
           "@types/bun": "latest",
+          typescript: "^7.0.2",
           ...(options.includeDb ? { "drizzle-kit": "^0.31.10" } : {}),
         },
       },

